kornia/augmentation/augmentation.py

```diff
@@ -1148,3 +1148,48 @@ def compute_transformation(self, input: torch.Tensor, params: Dict[str, torch.Te
 
     def apply_transform(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
         return F.apply_equalize(input, params)
+
+class GaussianBlur(AugmentationBase2D):
+    
+    r"""Apply gaussian blur given tensor image or a batch of tensor images randomly.
+
+    Args:
+        kernel_size (Tuple[int, int]): the size of the kernel.
+        sigma (Tuple[float, float]): the standard deviation of the kernel.
+        border_type (str): the padding mode to be applied before convolving.
+          The expected modes are: ``'constant'``, ``'reflect'``,
+          ``'replicate'`` or ``'circular'``. Default: ``'reflect'``.
+        return_transform (bool): if ``True`` return the matrix describing the transformation applied to each
+            input tensor. If ``False`` and the input is a tuple the applied transformation wont be concatenated.
+        same_on_batch (bool): apply the same transformation across the batch. Default: False.
+        p (float): probability of applying the transformation. Default value is 0.5.
+
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`, Optional: :math:`(B, 3, 3)`
+        - Output: :math:`(B, C, H, W)`
+
+    Note:
+        Input tensor must be float and normalized into [0, 1] for the best differentiability support.
+        Additionally, this function accepts another transformation tensor (:math:`(B, 3, 3)`), then the
+        applied transformation will be merged int to the input transformation tensor and returned.
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> input = torch.rand(1, 1, 5, 5)
+        >>> blur = GaussianBlur((3, 3), (0.1, 2.0), p=1.)
+        >>> blur(input)
+    """
+    def __init__(self, kernel_size: Tuple[int, int],
+                 sigma: Tuple[float, float],
+                 border_type: str = 'reflect', 
+                 return_transform: bool = False,
+                 same_on_batch: bool = False,
+                 p: float = 0.5) -> None:
+        super(GaussianBlur, self).__init__(p=p, return_transform=return_transform, same_on_batch=same_on_batch, p_batch=p)
+        self.transform = K.filters.GaussianBlur2d(kernel_size, sigma, border_type)
```

Comment:
We are making all parameters trainable. Please refer to other implementations for doing so.

Reply:
Can I ask what parameters should be trainable here? sigma?

Reply:
Theoretically, yes. If the user input sigma as ```nn.Parameter(1.)```, it will be trainable (the user may use ```xxx.backward()``` to update it if they want). Otherwise, it will be static in our current design.

Reply:
Thank you for your clarification. Can I ask if there are any other parameters should be trainable?

Reply:
Currently, only sigma shall be trainable since it is a continuous variable.

For kernel size and borderType, they are not going to be trainable for now since they are discrete variables. At some point, we did think about putting somewhat gradient estimation for the discrete variables (for augmentations), but I think it is not going to happen very soon. 

If you are interested in this, please feel free to have a PR or an issue to discuss more about it :)

Reply:
to make sigma to be trainable, some changes are needed here: https://github.com/kornia/kornia/blob/master/kornia/filters/kernels.py#L20-L25
but this should be done in a separated PR since might imply other changes I guess with typing. I would stick this PR to sigma with floats and leave for later the support to trainable tensors. We should consider later too to add too some sort of random gaussian blur/noise/etc.

Reply:
Ok, I'll resolve this for now, the reset changes to do is some more tests and the docs?